filament/src/UniformBuffer.h

```diff
@@ -66,15 +69,22 @@ class UniformBuffer { // NOLINT(cppcoreguidelines-pro-type-member-init)
 
     UniformBuffer& setUniforms(const UniformBuffer& rhs) noexcept;
 
-    // invalidate a range of uniforms and return a pointer to it. offset and size given in bytes
-    void* invalidateUniforms(size_t const offset, size_t const size) {
+    // Checks if a range needs to be invalidated by comparing the current value.
+    template<size_t Size>
+    bool invalidateNeeded(size_t const offset, void const* UTILS_RESTRICT v) const {
+        assert_invariant(offset + Size <= mSize);
+        void* const UTILS_RESTRICT addr = getUniformAddress(offset);
+        return bool(memcmp(addr, v, Size)); // inlined
+    }
+
+    // Invalidate a range of uniforms and return a pointer to it. offset and size given in bytes
+    void invalidateUniforms(size_t const offset, size_t const size) const {
```

Comment:
Doesn't seem like `offset` and `size` are actually used here.  Couldn't we move `mSomethingDirty` to `invalidate()` and just have that?

Reply:
ah yes, you're right, but there is a reason for it -- I wanted this internal API to be able to invalidate ranges; currently, it just invalidate the whole thing; but I wanted to keep that info.